50.005 Computer System Engineering/2018/Networking.md

```diff
@@ -0,0 +1,178 @@
+# Networking
+
+### Table of Contents
+- [Networking](#Networking)
+    - [Table of Contents](#Table-of-Contents)
+  - [The Internet - A nuts & bolts view](#The-Internet---A-nuts--bolts-view)
+  - [The Packet](#The-Packet)
+    - [Single Router Packet Delay Sources:](#Single-Router-Packet-Delay-Sources)
+    - [End-to-end Delay for many hosts](#End-to-end-Delay-for-many-hosts)
+    - [Average Link Utilization](#Average-Link-Utilization)
+    - [Modems](#Modems)
+  - [The Internet - A service view](#The-Internet---A-service-view)
+  - [Internet Protocol Stack (Layering)](#Internet-Protocol-Stack-Layering)
+    - [Internet Control Message Protocol (ICMP)](#Internet-Control-Message-Protocol-ICMP)
+    - [Internet Service Provider (ISP)](#Internet-Service-Provider-ISP)
+  - [Sharing of Information via Switching](#Sharing-of-Information-via-Switching)
+    - [Circuit Switching](#Circuit-Switching)
+    - [Packet Switching](#Packet-Switching)
+
+## The Internet - A nuts & bolts view
+1. Hosts (End Systems) running network apps
+2. Communication Links
+3. Packet Switches
+
+## The Packet
+- Packet = Header + Payload
+    - Header data is used by networking hardware to direct the packet to its destination
+    - Payload is extracted and used by application software.
+
+- Usually, the mode of transmission of packets is:
+`PC -> Switch -> Router -> Modem -> Modem -> Switch -> Servers`
+
+- When the packets arrive at the router, the packet arrival rate to link temporarily exceeds the output link capacity. Hence they would queue in the router buffers, causing a delay.
+    - If there are no free buffers for them to get into queue, then the packets will be dropped. This is known as packet loss.
+
+### Single Router Packet Delay Sources:
+
+![](https://i.imgur.com/V2iaWr9.png)
+Determined by `dnodal = dproc + dqueue + dtrans + dprop` 
+
+
+1. Nodal Processing *(dproc)*
+    - Examine packet header
+    - Check for bit errors 
+    - Determine output link (destination IP address in packet header) 
+2. Queueing *(dqueue)*
+    - Waiting time for packet to get to front of queue for output link
+    - Depends on congestion level (how much other users are also sending data)
+3. Transmission *(dtrans)*
+    - Time to push the whole packet *(all the bits)* from router to link 
+    - *dtrans* = packet length (bits) / link bandwith (bps)
+`dtrans = L/R` 
+4. Propagation *(dprop)* 
+    - Time for packet to move from beginning to end of the link
+    - *dprop* = length of physical link (m) / propagation speed in medium *(~2x10^8 m/s - 2/3 speed of light in vacuum)*
+`dprop = d/s` **m/s**
```

Comment:
The uh, formula and units be clearer. Now it looks awkward. Suggest placing everything inside code segment.

For formulas, you can try exporting HTML from https://www.codecogs.com/latex/eqneditor.php as well

<a href="https://www.codecogs.com/eqnedit.php?latex=\inline&space;\verb|dprop|&space;=&space;\frac{d}{s}&space;\mathrm{m/s}" target="_blank"><img src="https://latex.codecogs.com/gif.latex?\inline&space;\verb|dprop|&space;=&space;\frac{d}{s}&space;\mathrm{m/s}" title="\verb|dprop| = \frac{d}{s} \mathrm{m/s}" /></a>